src/data/sources/steam/Steam.vala

```diff
@@ -495,56 +507,266 @@ namespace GameHub.Data.Sources.Steam
+					collections_db.set_collection(key, collection);
+					created_collection = true;
+				}
+
+				if(!user_collections.get_object().has_member(key))
+				{
+					var object = new Json.Object();
+					object.set_string_member("id", key);
+					object.set_array_member("added", new Json.Array());
+					object.set_array_member("removed", new Json.Array());
+					user_collections.get_object().set_object_member(key, object);
+				}
+
+				var already_present = false;
+				user_collections.get_object().get_object_member(key).get_array_member("added").foreach_element((array, index, node) =>
+				{
+					if(node.get_int() == custom_appid_int) already_present = true;
+				});
+				if(!already_present)
 				{
-					tags_node.add_node(new BinaryVDF.StringNode.node((game.tags.index_of(tag) + 1).to_string(), tag.name));
+					user_collections.get_object().get_object_member(key).get_array_member("added").add_int_element(custom_appid_int);
 				}
 			}
 
-			game_node.add_node(tags_node);
+			debug(@"[Sources.Steam.set_shortcut_collection]\n$(Json.to_string(user_collections, true))");
 
-			root_node.add_node(game_node);
+			var generator = new Json.Generator();
+			generator.set_root(user_collections);
+			localconfig.get_object().get_object_member("UserLocalConfigStore").get_object_member("WebStorage").set_string_member("user-collections", generator.to_data(null));
+			FSUtils.write_string_to_file(FSUtils.file(config_dir.get_path(), "localconfig.vdf"), generate_vdf_from_json(localconfig.get_object()));
+			if(created_collection) collections_db.save(out error);
+			if(error != null) warning(@"[Sources.Steam.set_shortcut_collection] Error saving database: `%s`", error);
+		}
 
-			root_node.show();
+		// https://github.com/node-steam/vdf/blob/master/src/index.ts#L78-L117
+		private static string generate_vdf_from_json(Json.Object object, int level = 0)
+		{
+			var seperator = "	";
+			var result = "";
+			var indent = "";
 
-			BinaryVDF.write(shortcuts, root_node);
+			for(var i = 0; i < level; i++)
+			{
+				indent += seperator;
+			}
+
+			object.foreach_member((object, name, node) =>
+			{
+				if(node.get_node_type() == Json.NodeType.OBJECT && node.get_object() != null)
+				{
+					result += indent + "\"" + name + "\"\n" + indent + "{\n" + generate_vdf_from_json(node.get_object(), level + 1) + indent + "}\n";
+				}
+				else
+				{
+					var generator = new Json.Generator();
+					generator.set_root(node);
+					result += indent + "\"" + name + "\"" + seperator + seperator + generator.to_data(null) + "\n";
+				}
+			});
+
+			return result;
+		}
+
+		public static bool is_running()
+		{
+			if(Utils.run({"pidof", "steam"}).run_sync(true).exit_code == 0) return true;
+			return false;
+		}
+
+		public static string generate_new_appid(string exe, string name)
+		{
+			// https://github.com/boppreh/steamgrid/blob/master/games.go#L120
+			return (crc32(0, (exe + name).data) | 0x80000000).to_string();
 		}
```

Comment:
This is now replaced by a 4 byte little endian "appid" in the shortcut.vdf.

BigPicture is still using this tho. ((crc32(taget+name)|0x80000000)<<32|0x02000000)